packages/babel-generator/src/buffer.ts

```diff
@@ -469,18 +468,13 @@ export default class Buffer {
     cb();
   }
 
-  _normalizePosition(
-    prop: "start" | "end",
-    loc: Loc,
-    lineOffset: number,
-    columnOffset: number,
-  ) {
+  _normalizePosition(prop: "start" | "end", loc: Loc, columnOffset: number) {
     const pos = loc[prop];
     const target = this._sourcePosition;
 
     if (pos) {
-      target.line = pos.line + lineOffset;
-      target.column = pos.column + columnOffset;
+      target.line = pos.line;
+      target.column = Math.max(pos.column + columnOffset, 0);
```

Comment:
Can you add a note here to https://github.com/babel/babel/issues/15712? So that we can investigate this issue in the future.